include/mg_exceptions.hpp

```diff
@@ -105,4 +105,10 @@ struct SerializationException : public std::exception {
   const char *what() const noexcept override { return "Error in serialization!"; }
 };
 
+struct UnregisteredMemoryUsage : public std::exception {
+  const char *what() const noexcept override {
+    return "Thread not registered for memory usage. Please use MemoryDispatcher.";
```

Comment:
Maybe its better to say MemoryDispatcherGuard instead of MemoryDispatcher since thats what the user sees/uses